### DISCUSSION TITLE
The dropdowns and layouts in TailwindCSS doesn't working under Angular 10

### DISCUSSION THREAD

**Discussion by Ranzeplay** [tailwindlabs/tailwindcss#2605] [Help] [answered]
I used [@ikamva/ngx-tailwindcss](https://github.com/Applicafroguy/ngx-tailwindcss-schematic) to scaffold Tailwind CSS in Angular.
But when I build a Sidebar layout in the project, it becomes very weird. Like this:
![Screenshot 2020-10-18 120849](https://user-images.githubusercontent.com/69068007/96358528-b822dd00-113a-11eb-85a5-9a9beb667220.png)

I can't do anything with the dropdown, even close it

**Comment by adamwathan** [comment, marked as answer]
Hey! As mentioned on the Tailwind UI website, Tailwind UI is HTML-only, and you are expected to implement the JS necessary to make things like drop downs work yourself:

https://tailwindui.com/documentation#integrating-with-javascript-frameworks

**Comment by oschebesta** [comment]
You have to deal with opening and closing on your own. `*ngIf` could do the trick.
I am doing some research to get proper animations, for example with a transition-component like the one in headless-ui for react but only found this one [https://github.com/METACEO/ng-transition](https://github.com/METACEO/ng-transition). Are there any other ideas to get the in tailwindui described transitions to work with angular?

@adamwathan, you are doing great work with tailwind and all the other stuff, I really love it.

**Comment by abhishekpurecode** [comment]
When encountering issues with Tailwind CSS dropdowns and layouts in Angular 10, ensure proper integration and compatibility. Confirm that Tailwind classes are correctly applied in Angular components and templates. Check for potential conflicts with Angular's encapsulation mechanisms, and consider using the "::ng-deep" selector to address styling within Angular components. Additionally, verify the correct installation and configuration of Tailwind in your Angular project. By troubleshooting these aspects and ensuring seamless collaboration between Angular and Tailwind, you can resolve compatibility issues and create a harmonious blend of powerful styling and dynamic functionality in your web applications.

To know more, read here: https://purecode.ai/blogs/tailwind-dropdown/
